pkg/resiliency/error.go

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package resiliency
+
+import (
+	"fmt"
+)
+
+const (
+	none Tag = ""
+
+	// ResExt tracks unavailability of external resource (malicious or accidental)
+	ResExt Tag = "resExt"
+
+	// ResLimit tracks constraints on resources or API
+	ResLimit Tag = "resLimit"
+
+	// ResUnavailable tracks external manipulation (manual or accidental).
+	ResUnavailable Tag = "resUnavailable"
+
+	// ResDatapath tracks datapath misbehavior.
+	ResDatapath Tag = "resDatapath"
+
+	// ResControlplane tracks control plane misbehavior.
+	ResControlplane Tag = "resControlplane"
+
+	// ExtSys tracks failure of system interactions.
+	ExtSys Tag = "extSys"
+)
+
+type (
+	// Tag represent a resiliency issue category.
+	Tag string
+
+	taggedErr struct {
+		error
+		tag Tag
+	}
+
+	// RetryableErr tracks error types that could be retried.
+	RetryableErr struct {
+		taggedErr
+	}
+)
+
+// NewErr returns a new instance.
+func NewErr(e error, t Tag) taggedErr {
+	return taggedErr{
+		error: e,
+		tag:   t,
+	}
+}
+
+// Error returns a human representation.
+// TODO do we want to decorate at all??
```

Comment:
To kick off the discussion: What will the user do with this information?

Reply:
Removed!